### DISCUSSION TITLE
Is there a way to put these bars behind the circles?

### DISCUSSION THREAD

**Discussion by frederikhors** [tailwindlabs/tailwindcss#6148] [Help] [answered]
I couldn't understand how to put these bars **behind** the circles:

https://play.tailwindcss.com/peSieAptHf

![image](https://user-images.githubusercontent.com/41120635/142731919-a1cf51c1-b9b6-4cf4-b218-119febfc42dc.png)
I'm trying with `z-index` but it doesn't work!

Is there an alternative way?

```html
<div class="flex w-full p-20">
	<div class="w-full">
		<div class="flex mx-auto w-8 h-8 bg-gray-300 rounded-full"></div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="flex mx-auto w-8 h-8 bg-blue-300 rounded-full"></div>
			<div class="absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-green-500"></div>
			</div>
		</div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="mx-auto w-8 h-8 bg-blue-500 rounded-full"></div>
			<div class="absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-gray-700"></div>
			</div>
		</div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="mx-auto w-8 h-8 bg-green-500 rounded-full"></div>
			<div class="absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-red-500"></div>
			</div>
		</div>
	</div>
</div>
```

**Comment by dance2die** [comment, marked as answer]
You can set the negative z-indexes for lines.

With `jit` mode on, you can do `z-[-10]` for `z-index: 10px`.

And to keep the z-index contained within the current context, use [isolate](https://tailwindcss.com/docs/isolation#class-reference) to create a new context so `z-index` within it doesn't affect outside (not a requirement but if the site gets bigger, it can affect other places).

**demo**: https://play.tailwindcss.com/B6quvy7Lah

```html
<!--                            👇 -->
<div class="flex w-full p-20 isolate">
	<div class="w-full">
		<div class="flex mx-auto w-8 h-8 bg-gray-300 rounded-full"></div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="flex mx-auto w-8 h-8 bg-blue-300 rounded-full"></div>
			<div class="z-[-10] absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-green-500"></div>
			</div>
		</div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="mx-auto w-8 h-8 bg-blue-500 rounded-full"></div>
			<div class="z-[-10] absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-gray-700"></div>
			</div>
		</div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="mx-auto w-8 h-8 bg-green-500 rounded-full"></div>
			<div class="z-[-10] absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-red-500"></div>
			</div>
		</div>
	</div>
</div>
```

Instead of creating new DOM elements for the "bars", you can also use the new [before/after variants](https://blog.tailwindcss.com/tailwindcss-2-2#before-and-after-variants) to add lines between circles.

**Demo**: https://play.tailwindcss.com/XryQF1cu4T

```html
<div class="flex w-full p-20 isolate">
	<div class="w-full">
		<div class="flex mx-auto w-8 h-8 bg-gray-300 rounded-full"></div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="flex mx-auto w-8 h-8 bg-blue-300 rounded-full 
				after:absolute after:inset-0 after:top-1/2
				after:z-[-10] after:-translate-x-2/4 after:h-0.5
				after:bg-green-500
			"></div>
			<!-- <div class="absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-green-500"></div>
			</div> -->
		</div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="mx-auto w-8 h-8 bg-blue-500 rounded-full
				after:absolute after:inset-0 after:top-1/2
				after:z-[-10] after:-translate-x-2/4 after:h-0.5
				after:bg-gray-700
			"></div>
			<!-- <div class="absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-gray-700"></div>
			</div> -->
		</div>
	</div>

	<div class="w-full">
		<div class="relative">
			<div class="mx-auto w-8 h-8 bg-green-500 rounded-full
				after:absolute after:inset-0 after:top-1/2
				after:z-[-10] after:-translate-x-2/4 after:h-0.5
				after:bg-red-500"></div>
			<!-- <div class="absolute inset-0 flex items-center -translate-x-2/4">
				<div class="h-0.5 w-full bg-red-500"></div>
			</div> -->
		</div>
	</div>
</div>
``

**Reply by frederikhors** [reply]
Wow. What an answer! Sensational! Thank you so much!

**Reply by frederikhors** [reply]
Do you think there are any drawbacks or advantages to using "after" and "before" instead of creating the DOM elements?

**Comment by dance2die** [comment]
> Wow. What an answer! Sensational! Thank you so much!

You're welcome. Learned along the way myself as well :)

> Do you think there are any drawbacks or advantages to using "after" and "before" instead of creating the DOM elements?

Specific to your example, bars are used for decorations.  
- https://css-tricks.com/7-practical-uses-for-the-before-and-after-pseudo-elements-in-css/

So for the accessibility, it can work better 
- https://accessibleweb.com/question-answer/how-is-css-pseudo-content-treated-by-screen-readers/

You don't need to maintain extra DOM (without much semantics), which you might not remember what it is later on.

If the "bar" were to provide some purposes (clickable to show info, etc), then DOM version would work better as [pseudo elements cant' have event handlers attached](https://stackoverflow.com/a/9395877/4035) ([though there is a workaround, it's hacky](https://stackoverflow.com/a/39161931/4035)).
